src/Facets/CalldataVerificationFacet.sol

```diff
@@ -0,0 +1,240 @@
+    /// @param data The calldata to validate
+    /// @param dstCalldata The destination calldata to validate
+    /// @return isValid Whether the destination calldata is validate
+    function validateDestinationCalldata(
+        bytes calldata data,
+        bytes calldata dstCalldata
+    ) external pure returns (bool isValid) {
+        bytes4 selector = abi.decode(data, (bytes4));
+
+        // Case: Amarok
+        if (selector == AmarokFacet.startBridgeTokensViaAmarok.selector) {
+            (, AmarokFacet.AmarokData memory amarokData) = abi.decode(
+                data[4:],
+                (ILiFi.BridgeData, AmarokFacet.AmarokData)
+            );
+
+            return keccak256(dstCalldata) == keccak256(amarokData.callData);
+        }
+        if (
+            selector == AmarokFacet.swapAndStartBridgeTokensViaAmarok.selector
+        ) {
+            (, , AmarokFacet.AmarokData memory amarokData) = abi.decode(
+                data[4:],
+                (ILiFi.BridgeData, LibSwap.SwapData[], AmarokFacet.AmarokData)
+            );
+            return keccak256(dstCalldata) == keccak256(amarokData.callData);
+        }
+
+        // Case: Stargate
+        if (selector == StargateFacet.startBridgeTokensViaStargate.selector) {
+            (, StargateFacet.StargateData memory stargateData) = abi.decode(
+                data[4:],
+                (ILiFi.BridgeData, StargateFacet.StargateData)
+            );
+            return keccak256(dstCalldata) == keccak256(stargateData.callData);
+        }
+        if (
+            selector ==
+            StargateFacet.swapAndStartBridgeTokensViaStargate.selector
+        ) {
+            (, , StargateFacet.StargateData memory stargateData) = abi.decode(
+                data[4:],
+                (
+                    ILiFi.BridgeData,
+                    LibSwap.SwapData[],
+                    StargateFacet.StargateData
+                )
+            );
+            return keccak256(dstCalldata) == keccak256(stargateData.callData);
+        }
+        // Case: Celer
+        if (selector == CelerIMFacet.startBridgeTokensViaCelerIM.selector) {
+            (, CelerIMFacet.CelerIMData memory celerIMData) = abi.decode(
+                data[4:],
+                (ILiFi.BridgeData, CelerIMFacet.CelerIMData)
+            );
+            return keccak256(dstCalldata) == keccak256(celerIMData.callData);
+        }
+        if (
+            selector ==
+            CelerIMFacet.swapAndStartBridgeTokensViaCelerIM.selector
+        ) {
+            (, , CelerIMFacet.CelerIMData memory celerIMData) = abi.decode(
+                data[4:],
+                (
+                    ILiFi.BridgeData,
+                    LibSwap.SwapData[],
+                    CelerIMFacet.CelerIMData
+                )
+            );
+            return keccak256(dstCalldata) == keccak256(celerIMData.callData);
+        }
+
+        // All other cases
+        return false;
+    }
```

Comment:
We should add a function to parse calls to the `GenericSwapFacet` facet.

Reply:
@maxklenk What would we like to return from this?
```
reciever
amount
sendingAssetId
```

Reply:
@ezynda3 I think input and output are important
```
sendingAssetId <= first swap asset
sendingAmount <= first swap amount
receiver 
receivingAssetId <= last swap asset
receivingMinAmount <= passed minAmount
```